tools/internal-rules/no-useless-catch.js

```diff
@@ -0,0 +1,44 @@
+/**
```

Comment:
This feels like it shouldn't be an internal rule. It's useful in general, and not specific to ESLint. Granted, I'm not sure if it's important enough to be in the core, but maybe a separate plugin?

Reply:
Yeah, I would be fine with adding it as a core rule. I just made it an internal rule for now so that it doesn't need to wait for approval.